tutorials/preprocessing/35_artifact_correction_regression.py

```diff
@@ -23,146 +24,165 @@
-mne.viz.tight_layout()
-
-# plot regressed blink epochs
-fig = eog_epochs_clean.average(picks=plot_picks).plot(**evo_kwargs)
-fig.suptitle('EOG epochs, EOG regressed')
-mne.viz.tight_layout()
+# Create EOG evoked before regression
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Croft & Barry (2000) :footcite:`CroftBarry2000` suggest creating epochs based
+# on blink onsets and computing the evoked blink response. The averaging
+# procedure will suppress EEG signals that are not strictly time-locked with
+# the blink response. Ideally, one would create evokeds for both blinks and
+# saccades, and create two separate regression models. However, we will
+# restrict ourselves to just blink epochs, since MNE-Python contains an
+# automated method for creating those.
+
+eog_epochs = mne.preprocessing.create_eog_epochs(raw)
+# We need to explicitly specify that we want to average the EOG channel too.
+eog_evoked = eog_epochs.average('all')
+eog_evoked.plot('all')
+
+# perform regression on the evoked blink response
+model_evoked = EOGRegression(picks='eeg', picks_artifact='eog').fit(eog_evoked)
+model_evoked.plot(vmax=0.4)
+
+# apply the regression coefficients to the original epochs
+epochs_clean_evoked = model_evoked.apply(epochs)
+fig = epochs_clean_evoked.average('all').plot(**plot_kwargs)
+fig.set_size_inches(6, 6)
+
+# for good measure, also show the effect on the blink evoked
+model_evoked.apply(eog_evoked).plot('all')
 
 # %%
 # Visualize the effect on raw data
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-# We can also apply the regression directly to the raw data. To do this
-# following the Gratton method requires first computing the regression weights
-# *from epoched data with the evoked response subtracted out* (as we did
-# above). It is possible to instead compute regression weights directly from
-# the raw data, but this could bias the evoked response more than computing
-# the weights from data with the evoked response removed.
-
-# get ready to plot
-order = np.concatenate([  # plotting order: EOG+ECG first, then MEG
-    mne.pick_types(raw.info, meg=False, eog=True, ecg=True),
-    mne.pick_types(raw.info, meg=True)])
-raw_kwargs = dict(events=eog_epochs.events, order=order, start=20, duration=5,
-                  n_channels=40)
+# We can also apply the regression directly to the raw data. Here, we will use
+# the regression weights obtained from the blink evoked above and apply it to
+# the `~mne.io.Raw`.
+order = np.concatenate([  # plotting order: EOG first, then EEG
+    mne.pick_types(raw.info, meg=False, eog=True),
+    mne.pick_types(raw.info, eeg=True)])
+raw_kwargs = dict(events=eog_epochs.events, order=order, start=13, duration=3,
+                  n_channels=10, scalings=dict(eeg=50e-6, eog=250e-6))
 
 # plot original data
 raw.plot(**raw_kwargs)
 
 # regress (using betas computed above) & plot
-raw_clean, _ = mne.preprocessing.regress_artifact(raw, betas=betas)
+raw_clean = model_evoked.apply(raw)
 raw_clean.plot(**raw_kwargs)
```

Comment:
I think we should have a `.. note:` section somewhere to explain how it relates to SSP. Could say why and when SSP could be better? I have no strong opinion here.